util/types/time.go

```diff
@@ -274,34 +263,38 @@ func (t Time) ConvertToDuration() (Duration, error) {
 // Compare returns an integer comparing the time instant t to o.
 // If t is after o, return 1, equal o, return 0, before o, return -1.
 func (t Time) Compare(o Time) int {
+	return compareTime(t.Time, o.Time)
```

Comment:
Somewhere else need to compare `TimeInternal`, so I extract a `compareTime` function rather than 
make Compare a method of `Time`.